testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -1000,6 +1000,10 @@ void testCastToInterval(CastType castType, SqlOperatorFixture f) {
         "cast(-5723 as interval minute(4))",
         "-5723",
         "INTERVAL MINUTE(4) NOT NULL");
+    f.checkScalar(
```

Comment:
just integer?

Reply:
What else? I can add DECIMAL, like this:
```
    f.checkScalar(
        "cast(cast(1 as decimal) as interval minute)",
        "+1",
        "INTERVAL MINUTE NOT NULL");
```
Is it ok?
Approximate numerics (FLOAT, DOUBLE) are not supported for casting to interval.

Reply:
@alex-plekhanov Thanks, it would be helpful if you could add some more different types of tests you can think of

Reply:
Ok, thank you, I've added the test with DECIMAL.